### PR TITLE
Add setMetadataFilename method to Saml2MetadataFilter

### PR DESCRIPTION
**Pull request by GitHanter**
Add setMetadataFilename method to Saml2MetadataFilter

Populated shared objects RelyingPartyRegistrationRepository and DefaultRelyingPartyRegistrationResolver in Saml2LoginConfigurer class

Closes gh-9317

### CODE REVIEW

File: samples/boot/saml2login/src/main/java/sample/SecurityConfig.java [hunk 1]
```diff
@@ -33,4 +43,22 @@ RelyingPartyRegistrationRepository relyingPartyRegistrationRepository() {
 				.build();
 		return new InMemoryRelyingPartyRegistrationRepository(relyingPartyRegistration);
 	}
+
+	@Override
+	protected void configure(HttpSecurity http) throws Exception {
+		http.authorizeRequests(request -> request.anyRequest().authenticated())
+				.saml2Login(Customizer.withDefaults()).apply(
+				new SecurityConfigurerAdapter<DefaultSecurityFilterChain, HttpSecurity>() {
```
Comment (jzheaux) [inline samples/boot/saml2login/src/main/java/sample/SecurityConfig.java:51]: Let's simplify this to:

```java
@Bean 
Saml2MetadataFilter metadataFilter(RelyingPartyRegistrationRepository registrations) {
    DefaultRelyingPartyRegistrationResolver resolver = new DefaultRelyingPartyRegistrationResolver(registrations);
    return new Saml2MetadataFilter(resolver, new OpenSamlMetadataResolver());
}
```

I think the filter chain configuration can be removed at that point.

Also, in the sample, let's leave out the custom filename. The reason is that we want the samples to be a very simple introduction to Spring Security.

**Review by jzheaux — CHANGES_REQUESTED** [review]
Thanks, @GitHanter, for the PR! I've left my feedback inline.

**Comment by GitHanter** [comment]
@jzheaux 

Really appreciate your detail review, I'll resolve the issues you point out and force push two commits :

- one with `setMetadataFilename` method change
- another with HTTP header` Content-Disposition` encoding changes

**Comment by jzheaux** [comment]
@GitHanter are you able to make the requested changes? I'm happy to help make them if you are busy.

**Comment by GitHanter** [comment]
@jzheaux Sorry for the delay.

I was on my Spring Festival vacation last week, I'll do it this weekend CST.

File: saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/web/Saml2MetadataFilterTests.java [hunk 2]
```diff
@@ -120,4 +121,17 @@ public void setRequestMatcherWhenNullThenIllegalArgument() {
 		assertThatIllegalArgumentException().isThrownBy(() -> this.filter.setRequestMatcher(null));
 	}
 
+	@Test
```
Comment (jzheaux) [inline saml2/saml2-service-provider/src/test/java/org/springframework/security/saml2/provider/service/web/Saml2MetadataFilterTests.java:124]: Can you add a test that shows that when the metadata filename is set, then it's used when the filter is invoked?

File: saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/web/Saml2MetadataFilter.java [hunk 3]
```diff
@@ -17,6 +17,8 @@
 package org.springframework.security.saml2.provider.service.web;
```
Comment (jzheaux) [inline saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/web/Saml2MetadataFilter.java:17]: In any files you change, will you please make sure that the copyright is updated to the current year?

**Comment by jzheaux** [comment]
@GitHanter, thanks again for the PR! This is now merged into `master`. Also, I added a polish commit of 3e8ad4bc2bbbfef80180f95c803b32e5082f584e to simplify the test a bit.

**Pull request closed without merging**